retrieval/target.go

```diff
@@ -449,15 +450,29 @@ func (t *Target) scrape(appender storage.SampleAppender) (err error) {
 		},
 	}
 
-	var samples model.Vector
+	var (
+		samples       model.Vector
+		numOutOfOrder int
+	)
 	for {
 		if err = sdec.Decode(&samples); err != nil {
 			break
 		}
 		for _, s := range samples {
-			appender.Append(s)
+			err := appender.Append(s)
+			if err != nil {
+				if err == local.ErrOutOfOrderSample {
+					numOutOfOrder++
+				} else {
+					log.Warnf("Error inserting sample %v: %s", s, err)
+				}
+			}
+
 		}
 	}
+	if numOutOfOrder > 0 {
+		log.Warnf("Error on ingesting %d out-of-order samples")
```

Comment:
Which means basically going back to logging thousands of messages per scrape per target.
We should have a base logger per target so the log line will include `instance=...`. That should generally be sufficient to debug.

That would rather be part of my ever-pending refactoring.